src/OpenStudy/relations.jl

```diff
@@ -118,6 +119,109 @@ function _get_relation(source::String, target::String, relation_type::RelationTy
     return _RELATIONS[source][(target, relation_type)]
 end
 
+function _get_target_index_from_relation(
+    data::Data,
+    source::String,
+    source_index::Integer,
+    relation_attribute::String,
+)
+    source_element = data.raw[source][source_index]
+    target_reference_id = source_element[relation_attribute]
+    _, target_index = _get_index(data.data_index, target_reference_id)
+    return target_index
+end
+
+function _get_sources_indices_from_relations(
+    data::Data,
+    source::String,
+    target::String,
+    target_id::Integer,
+    relation_attribute::String,
+)
+    possible_elements = data.raw[source]
+
+    indices = Vector{Int32}()
+    for (index, element) in enumerate(possible_elements)
+        if haskey(element, relation_attribute)
+            if element[relation_attribute] == target_id
+                push!(indices, index)
+            end
+        end
+    end
+    return indices
+end
+
+function _get_element_related(data::Data, collection::String, index::Integer)
+    element = data.raw[collection][index]
+
+    # source_collection, target_collection, source_index, target_index
+    relations = Dict{Tuple{String,String,Int,Int},String}()
```

Comment:
@joaquimg We thought that we would use this dictionary to find the name of the attribute that represents a relation. It would help us later to know which attribute to remove when deleting a relation between two elements(given their collection name and indices). 

We might have forgotten to explain what was the element that the key is referencing in our comment.